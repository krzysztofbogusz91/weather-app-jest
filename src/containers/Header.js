import React from 'react';
import { Link } from 'react-router-dom'

export const Header = () =>{
    return(
        <div>
            <header className="m-4 header">
                <h1>WEATHER APP</h1>
            </header>
            <nav className="p-0 mt-4 navbar">
                <ul className="nav nav-tabs d-flex w-100">
                    <li className=" nav-item">
                        <Link to="/" className="nav-link" href="#">
                            Weather In Your Localisation
                        </Link>
                    </li>
                    <li className=" nav-item">
                        <Link to="/search" className="nav-link" href="#">
                            Search Weather
                        </Link>
                    </li>
                </ul>
            </nav>
      </div>
    )
}