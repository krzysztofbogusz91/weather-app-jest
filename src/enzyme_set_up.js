// setup file
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

export const config = configure({ adapter: new Adapter() });