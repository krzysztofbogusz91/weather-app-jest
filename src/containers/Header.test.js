import React from 'react';
import { shallow } from 'enzyme';
import { config } from '../enzyme_set_up';
import { Header } from './Header';

describe('Header', () => {
    const head = shallow(<Header />)
    it('should render properly', () => {
        expect(head).toMatchSnapshot()
    });
    it('should have class header', () => {
        expect(head.find('Link').exists()).toBe(true)
    });
});