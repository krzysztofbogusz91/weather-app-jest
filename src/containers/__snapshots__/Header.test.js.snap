// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="m-4 header"
>
          <h1>
                    WEATHER APP
          </h1>
</header>,
        <nav
          className="p-0 mt-4 navbar"
>
          <ul
                    className="nav nav-tabs d-flex w-100"
          >
                    <li
                              className=" nav-item"
                    >
                              <Link
                                        className="nav-link"
                                        href="#"
                                        replace={false}
                                        to="/"
                              >
                                        Weather In Your Localisation
                              </Link>
                    </li>
                    <li
                              className=" nav-item"
                    >
                              <Link
                                        className="nav-link"
                                        href="#"
                                        replace={false}
                                        to="/search"
                              >
                                        Search Weather
                              </Link>
                    </li>
          </ul>
</nav>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            WEATHER APP
</h1>,
          "className": "m-4 header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "WEATHER APP",
          },
          "ref": null,
          "rendered": "WEATHER APP",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="nav nav-tabs d-flex w-100"
>
            <li
                        className=" nav-item"
            >
                        <Link
                                    className="nav-link"
                                    href="#"
                                    replace={false}
                                    to="/"
                        >
                                    Weather In Your Localisation
                        </Link>
            </li>
            <li
                        className=" nav-item"
            >
                        <Link
                                    className="nav-link"
                                    href="#"
                                    replace={false}
                                    to="/search"
                        >
                                    Search Weather
                        </Link>
            </li>
</ul>,
          "className": "p-0 mt-4 navbar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className=" nav-item"
>
                <Link
                                className="nav-link"
                                href="#"
                                replace={false}
                                to="/"
                >
                                Weather In Your Localisation
                </Link>
</li>,
              <li
                className=" nav-item"
>
                <Link
                                className="nav-link"
                                href="#"
                                replace={false}
                                to="/search"
                >
                                Search Weather
                </Link>
</li>,
            ],
            "className": "nav nav-tabs d-flex w-100",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="nav-link"
                  href="#"
                  replace={false}
                  to="/"
>
                  Weather In Your Localisation
</Link>,
                "className": " nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Weather In Your Localisation",
                  "className": "nav-link",
                  "href": "#",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Weather In Your Localisation",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="nav-link"
                  href="#"
                  replace={false}
                  to="/search"
>
                  Search Weather
</Link>,
                "className": " nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Search Weather",
                  "className": "nav-link",
                  "href": "#",
                  "replace": false,
                  "to": "/search",
                },
                "ref": null,
                "rendered": "Search Weather",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "nav",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="m-4 header"
>
            <h1>
                        WEATHER APP
            </h1>
</header>,
          <nav
            className="p-0 mt-4 navbar"
>
            <ul
                        className="nav nav-tabs d-flex w-100"
            >
                        <li
                                    className=" nav-item"
                        >
                                    <Link
                                                className="nav-link"
                                                href="#"
                                                replace={false}
                                                to="/"
                                    >
                                                Weather In Your Localisation
                                    </Link>
                        </li>
                        <li
                                    className=" nav-item"
                        >
                                    <Link
                                                className="nav-link"
                                                href="#"
                                                replace={false}
                                                to="/search"
                                    >
                                                Search Weather
                                    </Link>
                        </li>
            </ul>
</nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              WEATHER APP
</h1>,
            "className": "m-4 header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "WEATHER APP",
            },
            "ref": null,
            "rendered": "WEATHER APP",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="nav nav-tabs d-flex w-100"
>
              <li
                            className=" nav-item"
              >
                            <Link
                                          className="nav-link"
                                          href="#"
                                          replace={false}
                                          to="/"
                            >
                                          Weather In Your Localisation
                            </Link>
              </li>
              <li
                            className=" nav-item"
              >
                            <Link
                                          className="nav-link"
                                          href="#"
                                          replace={false}
                                          to="/search"
                            >
                                          Search Weather
                            </Link>
              </li>
</ul>,
            "className": "p-0 mt-4 navbar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className=" nav-item"
>
                  <Link
                                    className="nav-link"
                                    href="#"
                                    replace={false}
                                    to="/"
                  >
                                    Weather In Your Localisation
                  </Link>
</li>,
                <li
                  className=" nav-item"
>
                  <Link
                                    className="nav-link"
                                    href="#"
                                    replace={false}
                                    to="/search"
                  >
                                    Search Weather
                  </Link>
</li>,
              ],
              "className": "nav nav-tabs d-flex w-100",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link"
                    href="#"
                    replace={false}
                    to="/"
>
                    Weather In Your Localisation
</Link>,
                  "className": " nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Weather In Your Localisation",
                    "className": "nav-link",
                    "href": "#",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Weather In Your Localisation",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link"
                    href="#"
                    replace={false}
                    to="/search"
>
                    Search Weather
</Link>,
                  "className": " nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Search Weather",
                    "className": "nav-link",
                    "href": "#",
                    "replace": false,
                    "to": "/search",
                  },
                  "ref": null,
                  "rendered": "Search Weather",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "nav",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
